soc/arm/atmel_sam0/samd21/soc.h

```diff
@@ -68,4 +68,21 @@
 #define SOC_ATMEL_SAM0_APBB_FREQ_HZ SOC_ATMEL_SAM0_MCK_FREQ_HZ
 #define SOC_ATMEL_SAM0_APBC_FREQ_HZ SOC_ATMEL_SAM0_MCK_FREQ_HZ
 
+
+#define _SOC_ATMEL_SAM0_SERCOM_CHECK(dt_index, type, sercom_id) \
+	(IS_ENABLED(DT_ATMEL_SAM0_ ## type ## _ ## dt_index) && \
+	(DT_ATMEL_SAM0_ ## type ## _ ## dt_index ## _SERCOM_ID) == sercom_id)
+#define SOC_ATMEL_SAM0_SERCOM_IS_ENABLED(type, id) (	\
```

Comment:
I think this should be moved to `atmel_sam0/common/soc_sercom.h`

Reply:
I've changed it to use direct DTS aliases, which removes the need for those macros entirely.  However, this means supporting SAME54 depends on #14897 to support the extra interrupts.